### PR TITLE
Admin API Integration Tests

### PR DESCRIPTION
**Pull request by rashm1n**
Integration test cases  of APIDenyPolicyTestCase and APISystemScopesTestCase

### CODE REVIEW

File: modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/helpers/AdminApiTestHelper.java [hunk 1]
```diff
@@ -18,14 +18,7 @@
...
 import org.testng.Assert;
-import org.wso2.am.integration.clients.admin.api.dto.AdvancedThrottlePolicyDTO;
-import org.wso2.am.integration.clients.admin.api.dto.APICategoryDTO;
-import org.wso2.am.integration.clients.admin.api.dto.ApplicationThrottlePolicyDTO;
-import org.wso2.am.integration.clients.admin.api.dto.CustomRuleDTO;
-import org.wso2.am.integration.clients.admin.api.dto.EnvironmentDTO;
-import org.wso2.am.integration.clients.admin.api.dto.LabelDTO;
-import org.wso2.am.integration.clients.admin.api.dto.SubscriptionThrottlePolicyDTO;
-import org.wso2.am.integration.clients.admin.api.dto.VHostDTO;
+import org.wso2.am.integration.clients.admin.api.dto.*;
```
Comment (CrowleyRajapakse) [inline modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/helpers/AdminApiTestHelper.java:21]: let's remove those automatic * imports

File: modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/impl/RestAPIAdminImpl.java [hunk 2]
```diff
@@ -21,48 +21,9 @@
...
-import org.wso2.am.integration.clients.admin.api.dto.KeyManagerDTO;
-import org.wso2.am.integration.clients.admin.api.dto.KeyManagerListDTO;
-import org.wso2.am.integration.clients.admin.api.dto.LabelDTO;
-import org.wso2.am.integration.clients.admin.api.dto.LabelListDTO;
-import org.wso2.am.integration.clients.admin.api.dto.SettingsDTO;
-import org.wso2.am.integration.clients.admin.api.dto.SubscriptionThrottlePolicyDTO;
-import org.wso2.am.integration.clients.admin.api.dto.WorkflowDTO;
-import org.wso2.am.integration.clients.admin.api.dto.WorkflowInfoDTO;
-import org.wso2.am.integration.clients.admin.api.dto.WorkflowListDTO;
+import org.wso2.am.integration.clients.admin.api.*;
```
Comment (CrowleyRajapakse) [inline modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/impl/RestAPIAdminImpl.java:24]: need to remove start imports here as well

File: modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/impl/RestAPIAdminImpl.java [hunk 3]
```diff
@@ -435,6 +466,23 @@ public SettingsDTO getSettings() throws ApiException {
...
+    /**
+     * Updates an deny throttling policy.
+     *
+     * @param conditionId policy id of the deny throttling policy to be updated.
+     * @param conditionType condition type of the deny throttling policy to be.
+     * @param blockingConditionStatusDTO deny throttling policy status DTO to be updated.
+     * @return API response returned by API call.
+     * @throws ApiException if an error occurs while creating the deny throttling policy.
+     */
+    public ApiResponse<BlockingConditionDTO> updateDenyThrottlingPolicy(String conditionId, String conditionType, BlockingConditionStatusDTO blockingConditionStatusDTO) throws ApiException {
```
Comment (RAVEENSR) [inline modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/impl/RestAPIAdminImpl.java:478]: Looks line in here line limit(120) exceeds?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java [hunk 4]
```diff
@@ -0,0 +1,457 @@
+package org.wso2.am.integration.tests.other;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpStatus;
+import org.testng.Assert;
+import org.testng.annotations.*;
```
Comment (RAVEENSR) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java:7]: Avoid wild card imports

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+import org.wso2.am.integration.test.helpers.AdminApiTestHelper;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationConstants;
+import org.wso2.am.integration.test.utils.bean.APIRequest;
+import org.wso2.am.integration.tests.api.lifecycle.APIManagerLifecycleBaseTest;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.test.utils.http.client.HttpRequestUtil;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+
+import java.net.URL;
+import java.util.*;
```
Comment (RAVEENSR) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java:24]: Avoid wild card imports

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java [hunk 6]
```diff
@@ -0,0 +1,457 @@
...
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        for (String conditionId : addedPolicyIds) {
+            restAPIAdmin.deleteDenyThrottlingPolicy(conditionId);
+        }
+        super.cleanUp();
+    }
+
+}
```
Comment (RAVEENSR) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java:457]: Add a new line at the end of the file

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java (hunk 4 above)
Comment (rashm1n) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java:7]: ack

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java (hunk 5 above)
Comment (rashm1n) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java:24]: Ack

File: modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/impl/RestAPIAdminImpl.java (hunk 3 above)
Comment (rashm1n) [reply modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/impl/RestAPIAdminImpl.java:478]: Refactored

**Comment by CrowleyRajapakse** [comment]
APIDenyPolicyTestCase related some test cases seems to be failing in the report, let's check and fix

**Review by CrowleyRajapakse — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java [hunk 7]
```diff
@@ -0,0 +1,362 @@
+package org.wso2.am.integration.tests.other;
```
Comment (malinthaprasan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java:1]: License missing

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+        Object conditionValue = "/denyNonExisting/1.0.0/";
+
+        BlockingConditionDTO blockingConditionDTO = new BlockingConditionDTO();
+        blockingConditionDTO.setConditionStatus(conditionStatus);
+        blockingConditionDTO.setConditionValue(conditionValue);
+        blockingConditionDTO.setConditionType(BlockingConditionDTO.ConditionTypeEnum.API);
+        try {
+            ApiResponse<BlockingConditionDTO> addedPolicy = restAPIAdmin.addDenyThrottlingPolicy(blockingConditionDTO);
+        } catch (ApiException e) {
+            Assert.assertEquals(e.getCode(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
```
Comment (malinthaprasan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java:120]: Does this give a 500? Ideally it should give 400.. 
If 500, please create an issue, and mention about this test case as well. So if someone fixing the issue, he can fix the test as well.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/admin/APISystemScopesTestCase.java [hunk 9]
```diff
@@ -0,0 +1,103 @@
+package org.wso2.am.integration.tests.restapi.admin;
```
Comment (malinthaprasan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/admin/APISystemScopesTestCase.java:1]: Add License header

**Review by malinthaprasan — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java (hunk 8 above)
Comment (rashm1n) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java:120]: Yes It gives 500 not 400. Will create an issue.

Comment (rashm1n) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java:120]: https://github.com/wso2/product-apim/issues/11213

**Comment by chamilaadhi** [comment]
@rashm1n plz check and merge

**Pull request merged**